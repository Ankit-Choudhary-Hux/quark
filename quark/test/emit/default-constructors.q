class A {
    String name;
    A(String name) {
        self.name = name;
    }
}

class B extends A {
    void greet() {
        // XXX: self is required, do we want that?
        print("Hello, my name is " + self.name);
    }
}

class C extends A {
    C(String name) {
	super("C" + name);
    }
    void greet() {
	print("Greetings, my name is " + self.name);
    }
}

void main() {
    B b = new B("Bob");
    b.greet();
    C c = new C("arole");
    c.greet();
}
