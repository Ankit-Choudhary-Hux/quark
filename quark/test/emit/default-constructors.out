Hello, my name is Bob
Greetings, my name is Carole
